river/Config.zig

```diff
@@ -196,16 +205,48 @@ pub fn shouldFloat(self: Self, view: *View) bool {
 
 pub fn csdAllowed(self: Self, view: *View) bool {
     if (view.getAppId()) |app_id| {
-        if (self.csd_filter_app_ids.contains(std.mem.span(app_id))) {
+        if (self.matchesFilterPattern(FilterKind.@"app-id", FilterEffect.csd, std.mem.span(app_id))) {
             return true;
         }
     }
 
     if (view.getTitle()) |title| {
-        if (self.csd_filter_titles.contains(std.mem.span(title))) {
+        if (self.matchesFilterPattern(FilterKind.title, FilterEffect.csd, std.mem.span(title))) {
             return true;
         }
     }
 
     return false;
 }
+
+fn matchesFilterPattern(self: Self, kind: FilterKind, effect: FilterEffect, item: [:0]const u8) bool {
+    const filters = switch (effect) {
+        .float => switch (kind) {
+            .title => self.float_filter_titles,
+            .@"app-id" => self.float_filter_app_ids,
+        },
+        .csd => switch (kind) {
+            .title => self.csd_filter_titles,
+            .@"app-id" => self.csd_filter_app_ids,
+        },
+    };
+    var it = filters.keyIterator();
```

Comment:
Originally, the lists were kept separate to that `filters.contains()` could be used. This is no longer the case, so can join them all into a single list, which should make the code a bit more straight-forward. Just create a simple struct which points to a string, and also has booth `FilterKind` and `FilterEffect` field. I'd personally also add an arena which owns the strings for simplicity. Then add an ArrayList of that struct (not owned by the arena, growables don't go well with those).

Reply:
Yes, I can see how this makes sense, however I didn't understand "arena" concept. If you could point me to resource or example of it's use, I would consider implementing it.

Reply:
You can ignore the arena bit. I forgot that filters can also be removed, so the strings need to be free-able, meaning they need to be owned by gpa.